sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -959,38 +963,53 @@ case class SortMergeJoinExec(
     // Scan the left and right buffers to find all matched rows.
     val matchRowsInBuffer =
       s"""
-         |int $leftIndex;
-         |int $rightIndex;
-         |
-         |for ($leftIndex = 0; $leftIndex < $leftBuffer.size(); $leftIndex++) {
+         |while ($leftIndex < $leftBuffer.size()) {
          |  $leftOutputRow = (InternalRow) $leftBuffer.get($leftIndex);
-         |  for ($rightIndex = 0; $rightIndex < $rightBuffer.size(); $rightIndex++) {
+         |  while ($rightIndex < $rightBuffer.size()) {
```

Comment:
Why do you need replace for with while?

Reply:
Convenient control, go to +1 in different places.